src/components/ContentSidebar/ActivityFeed/task/Task.js

```diff
@@ -0,0 +1,143 @@
+/**
+ * @flow
+ * @file Tasks component
+ */
+
+import React, { Component } from 'react';
+import type { Node } from 'react';
+import { FormattedDate, FormattedMessage } from 'react-intl';
+
+import Comment from '../comment';
+import CompletedAssignment from './CompletedAssignment';
+import messages from '../../../messages';
+import PendingAssignment from './PendingAssignment';
+import RejectedAssignment from './RejectedAssignment';
+import type { ActionItemError, User } from '../../../../flowTypes';
+import type { Comments, Contacts, InputState, Tasks, Translations, Versions } from '../activityFeedFlowTypes';
+
+import './Task.scss';
+
+const TASK_APPROVED = 'approved';
+const TASK_REJECTED = 'rejected';
+const TASK_COMPLETED = 'completed';
+const TASK_INCOMPLETE = 'incomplete';
+
+type Props = {
+    assignees: Array<{
+        id: number,
+        user: User,
+        status: string
+    }>,
+    createdAt: number | string,
+    createdBy: User,
+    currentUser: User,
+    dueDate: any,
+    error: ActionItemError,
+    handlers: {
+        comments?: Comments,
+        tasks?: Tasks,
+        contacts?: Contacts,
+        versions?: Versions
+    },
+    id: string,
+    inputState: InputState,
+    isPending: boolean,
+    onDelete: Function,
+    onEdit: Function,
+    onTaskAssignmentUpdate: Function,
+    permissions: {
+        comment_delete: boolean,
+        comment_edit: boolean
+    },
+    translatedTaggedMessage: string,
+    translations: Translations,
+    taggedMessage: string
+};
+
+// eslint-disable-next-line
+class Task extends Component<Props> {
```

Comment:
Can these extend PureComponent instead?

Reply:
I'll make a tech foundation ticket to do that